### PR TITLE
Added density_scatter_plot

### PR DESCRIPTION
**Pull request by ptajvar**
Adding the function density_scatter_plot that creates scatter plot(s) of abundance data for two markers in one or multiple conditions. The points in the scatter plot are colored based on region density. 

Fixes: exe-1830

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

New test functions are added to test the function with different input combinations. 

## PR checklist:

- [x] This comment contains a description of changes (with reason).
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and documentation and corrected any misspellings
- [x] I have documented any significant changes to the code in [CHANGELOG.md](../CHANGELOG.md)

### CODE REVIEW

File: src/pixelator/plot/__init__.py [hunk 1]
```diff
@@ -935,6 +939,165 @@ def plot_colocalization_diff_volcano(
     return fig, ax
 
 
+def __plot_joint_distribution(data, x, y, **kargs):
```
Comment (johandahlberg) [inline src/pixelator/plot/__init__.py:942]: I think perhaps these should just have a single underscore.

**Review by johandahlberg — APPROVED** [review]
Minor comment. But looks good to me otherwise.

File: src/pixelator/plot/__init__.py [hunk 2]
```diff
@@ -935,6 +939,165 @@ def plot_colocalization_diff_volcano(
...
+
+    if facet_column is not None or facet_row is not None:
+        plot_grid = sns.FacetGrid(data=data, col=facet_column, row=facet_row)
+        plot_grid.map_dataframe(
+            sns.scatterplot,
+            x=marker1,
+            y=marker2,
+            hue="density",
+            palette=jet_colormap,
+            size=0.1,
```
[the remark this answers is not in the document]
Comment (maxkarlsson) [reply src/pixelator/plot/__init__.py:1080]: Take these suggestions as mere suggestions! I don't think we _have_ to add them - just want to pick your brain

File: src/pixelator/plot/__init__.py [hunk 3]
```diff
@@ -935,6 +939,165 @@ def plot_colocalization_diff_volcano(
...
+        )
+        if gate is not None:
```
Comment (maxkarlsson) [inline src/pixelator/plot/__init__.py:1082]: Looks like it worked out really elegant with the facet-wise gates!

File: src/pixelator/plot/__init__.py [hunk 4]
```diff
@@ -935,6 +939,165 @@ def plot_colocalization_diff_volcano(
     return fig, ax
 
 
+def __plot_joint_distribution(data, x, y, **kargs):
+    g = sns.JointGrid(data, x=x, y=y)
+    g.plot_marginals(sns.kdeplot)
+    g.plot_joint(
```
Comment (maxkarlsson) [inline src/pixelator/plot/__init__.py:945]: Style suggestion: Is it possible to remove the axes of the marginal plots, and use fill color instead of stroke color?

File: src/pixelator/plot/__init__.py [hunk 5]
```diff
@@ -935,6 +939,165 @@ def plot_colocalization_diff_volcano(
...
+        if gate is not None:
+            plot_grid.map_dataframe(
+                __add_gate_box,
+                gate=gate,
+                marker1=marker1,
+                marker2=marker2,
+                facet_row=facet_row,
+                facet_column=facet_column,
+            )
+    else:
```
Comment (maxkarlsson) [inline src/pixelator/plot/__init__.py:1091]: Is it possible to remove marginal densities when plotting a single facet? I think some users might not want to use them every time

File: src/pixelator/plot/constants.py [hunk 6]
```diff
@@ -1,6 +1,6 @@
 """Contains constants used as part of plotting modules.
 
-Copyright (c) 2023 Pixelgen Technologies AB.
+Copyright © 2023 Pixelgen Technologies AB.
```
Comment (maxkarlsson) [inline src/pixelator/plot/constants.py:3]: Is this change intentional?

**Review by maxkarlsson — COMMENTED** [review]
I think this looks great! I have some comments, mostly suggestions for the documentation, and some comments regarding the plot style.

File: src/pixelator/plot/constants.py (hunk 6 above)
Comment (ptajvar) [reply src/pixelator/plot/constants.py:3]: Yes, the pre-commit hook requires this.

File: src/pixelator/plot/__init__.py (hunk 4 above)
Comment (ptajvar) [reply src/pixelator/plot/__init__.py:947]: Done.

File: src/pixelator/plot/__init__.py (hunk 5 above)
Comment (ptajvar) [reply src/pixelator/plot/__init__.py:1101]: Done, show_marginal is now an input and defaults to false.

File: src/pixelator/plot/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ptajvar) [reply src/pixelator/plot/__init__.py:1089]: size doesn't seem to affect the actual dot sizes as long as they are all the same value.
